### PR TITLE
ci: override -vm-size to run the tests on GPU machines

### PR DESCRIPTION
**Pull request by kzys**
### Change Summary

What and Why:

GPU machines have some quirks (e.g. use Cloud Hypervisor instead of Firecracker). While most of the tests don't need GPU, running these tests would increase our coverage.

How:

Related to:

---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team
- [x] n/a

### CODE REVIEW

**Review by benbjohnson — APPROVED** [review]

File: test/preflight/testlib/test_env.go [hunk 1]
```diff
@@ -180,6 +181,10 @@ type testingTWrapper interface {
 
 // Fly runs a flyctl the result
 func (f *FlyctlTestEnv) Fly(flyctlCmd string, vals ...interface{}) *FlyctlResult {
+	if strings.HasPrefix(flyctlCmd, "machine run ") || strings.HasPrefix(flyctlCmd, "launch ") {
+		flyctlCmd += " --vm-gpu-kind a100-sxm4-80gb --vm-cpu-kind performance --vm-memory 8192 "
```
Comment (dangra) [inline test/preflight/testlib/test_env.go:185]: A few notes:

* `a100-sxm4-80gb` are scarse
* `a100-pcie-40gb` or `l40s` availability are more common but only on `iad` and `ord` regions
* `--vm-size a100-40gb` or `--vm-size l40s` are useful shortcuts

File: .github/workflows/preflight.yml [hunk 2]
```diff
@@ -41,9 +42,11 @@ jobs:
         env:
           FLY_PREFLIGHT_TEST_ACCESS_TOKEN: ${{ secrets.FLYCTL_PREFLIGHT_CI_FLY_API_TOKEN }}
           FLY_PREFLIGHT_TEST_FLY_ORG: flyctl-ci-preflight
-          FLY_PREFLIGHT_TEST_FLY_REGIONS: iad syd
+          # This VM size is only available in ORD.
+          FLY_PREFLIGHT_TEST_FLY_REGIONS: ord
```
Comment (dangra) [inline .github/workflows/preflight.yml:46]: some tests relies on the secondary region to be set. Is this disabling those preflight tests forever?

File: test/preflight/apps_v2_integration_test.go [hunk 3]
```diff
@@ -455,16 +456,27 @@ func TestErrOutput(t *testing.T) {
 	res = f.FlyAllowExitFailure("machine update --vm-memory 10 %s --yes", firstMachine.ID)
 	require.Contains(f, res.StdErrString(), "invalid memory size")
 
-	f.Fly("machine update --vm-cpus 4 %s --vm-memory 2048 --yes", firstMachine.ID)
+	// This should fail on GPU machines because they're performance VMs.
+	if f.IsGpuMachine() {
+		res = f.FlyAllowExitFailure("machine update --vm-cpus 4 %s --vm-memory 2048 --yes", firstMachine.ID)
+		require.Contains(f, res.StdErrString(), "memory size for config is too low")
```
Comment (kzys) [inline test/preflight/apps_v2_integration_test.go:462]: Nit: You can `return` from here since the rest is having `if ! f.isGpuMachine()`

**Review by billyb2 — APPROVED** [review]
LGTM

**Pull request merged**